src/silo/config/preprocessing_config.cpp

```diff
@@ -53,102 +159,80 @@ std::optional<std::filesystem::path> PreprocessingConfig::getNdjsonInputFilename
-         INTERMEDIATE_RESULTS_DIRECTORY_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      intermediate_results_directory = *value;
+   if (auto var = config_source.getPath(intermediateResultsDirectoryOptionKey())) {
+      intermediate_results_directory = var.value();
    }
-   if (auto value = config.getString(PREPROCESSING_DATABASE_LOCATION_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         PREPROCESSING_DATABASE_LOCATION_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      preprocessing_database_location = *value;
+   if (auto var = config_source.getPath(preprocessingDatabaseLocationOptionKey())) {
+      preprocessing_database_location = var.value();
    }
-   if (auto value = config.getUInt32(DUCKDB_MEMORY_LIMIT_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         DUCKDB_MEMORY_LIMIT_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      duckdb_memory_limit_in_g = value;
+   if (auto var = config_source.getUint32(duckdbMemoryLimitInGTimeOptionKey())) {
+      duckdb_memory_limit_in_g = var.value();
    }
-   if (auto value = config.getString(LINEAGE_DEFINITIONS_FILENAME_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         LINEAGE_DEFINITIONS_FILENAME_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      lineage_definitions_file = *value;
+   if (auto var = config_source.getPath(lineageDefinitionsFilenameOptionKey())) {
+      lineage_definitions_file = var.value();
    }
-   if (auto value = config.getString(NDJSON_INPUT_FILENAME_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         NDJSON_INPUT_FILENAME_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      ndjson_input_filename = *value;
+   if (auto var = config_source.getPath(ndjsonInputFilenameOptionKey())) {
+      ndjson_input_filename = var.value();
    }
-   if (auto value = config.getString(REFERENCE_GENOME_FILENAME_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         REFERENCE_GENOME_FILENAME_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      reference_genome_file = *value;
+   if (auto var = config_source.getPath(databaseConfigFileOptionKey())) {
+      database_config_file = var.value();
    }
+   if (auto var = config_source.getPath(referenceGenomeFilenameOptionKey())) {
+      reference_genome_file = var.value();
+   }
+}
+
+std::vector<std::filesystem::path> PreprocessingConfig::getConfigPaths() const {
+   std::vector<std::filesystem::path> result;
+   if (default_preprocessing_config.has_value()) {
+      result.emplace_back();
+   }
+   if (preprocessing_config.has_value()) {
+      result.emplace_back();
+   }
+   return result;
 }
 
 }  // namespace silo::config
 
+namespace silo::common {
```

Comment:
toDebugString here is one definition of various overloads. If this definition is only used here then perhaps anonymous would still work; but I also think it is confusing to have an overloaded function that doesn't have all of the same definitions in all places. To be fair, it may not be properly public, either, so perhaps that should be done?